ledger/participant-integration-api/src/main/scala/platform/store/backend/oracle/OracleStorageBackend.scala

```diff
@@ -0,0 +1,520 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.backend.oracle
+
+import anorm.SqlParser.get
+import anorm.{SQL, SqlStringInterpolation}
+import com.daml.ledger.api.v1.command_completion_service.CompletionStreamResponse
+import com.daml.ledger.participant.state.v1.Offset
+import com.daml.ledger.{ApplicationId, TransactionId}
+import com.daml.lf.data.Ref
+import com.daml.lf.data.Ref.Party
+import com.daml.platform.store.appendonlydao.events.{ContractId, EventsTable, Key, Raw}
+import com.daml.platform.store.backend.common.{
+  AppendOnlySchema,
+  CommonStorageBackend,
+  TemplatedStorageBackend,
+}
+import com.daml.platform.store.backend.{DbDto, StorageBackend}
+
+import java.sql.Connection
+import java.time.Instant
+
+private[backend] object OracleStorageBackend
+    extends StorageBackend[AppendOnlySchema.Batch]
+    with CommonStorageBackend[AppendOnlySchema.Batch] {
+
+  override def reset(connection: Connection): Unit =
+    List(
+      "truncate table configuration_entries cascade",
+      "truncate table package_entries cascade",
+      "truncate table parameters cascade",
+      "truncate table participant_command_completions cascade",
+      "truncate table participant_command_submissions cascade",
+      "truncate table participant_events_divulgence cascade",
+      "truncate table participant_events_create cascade",
+      "truncate table participant_events_consuming_exercise cascade",
+      "truncate table participant_events_non_consuming_exercise cascade",
+      "truncate table parties cascade",
+      "truncate table party_entries cascade",
+    ).map(SQL(_)).foreach(_.execute()(connection))
+
+  override def enforceSynchronousCommit(connection: Connection): Unit = () // Not supported
+
+  override def duplicateKeyError: String = "unique constraint"
+
+  val SQL_INSERT_COMMAND: String =
```

Comment:
is this the same SQL literal than what we have for H2?
if yes, can you add here a // TODO appen-only: reuse -- this is the same as the H2 counterpart

Reply:
No the Oracle error string is different so best to keep them separate